SlowCheetah.VisualStudio/Utilities/PackageUtilities.cs

```diff
@@ -0,0 +1,147 @@
+        /// <returns>The validated and culled list of extensions. The returned items all begin with a ".".</returns>
+        public static string[] GetExtensionList(string items)
+        {
+            const string extensionPrefix = ".";
+            List<string> list = new List<string>();
+            if (!string.IsNullOrWhiteSpace(items))
+            {
+                string[] extensions = items.Split(';');
+                foreach (string extension in extensions)
+                {
+                    string extensionValue = string.IsNullOrWhiteSpace(extension) ? string.Empty : extension.Trim();
+                    if (extensionValue != string.Empty)
+                    {
+                        if (!extensionValue.StartsWith(extensionPrefix))
+                        {
+                            extensionValue = extensionPrefix + extensionValue;
+                        }
+
+                        list.Add(extensionValue);
+                    }
+                }
+            }
+            return list.ToArray();
+        }
+
+        /// <summary>
+        /// Gets a list of unsupported transformation extensions,
+        /// each starting with a ".".
+        /// </summary>
+        /// <returns>The list of extensions</returns>
+        public static List<string> GetExcludedExtensions()
+        {
+            return ExcludedExtensions;
+        }
+
+        /// <summary>
+        /// Verifies if the extension of the given file is supported
+        /// </summary>
+        /// <param name="filepath">Full path to the file</param>
+        /// <returns>True if the file is supported</returns>
+        public static bool IsExtensionSupportedForFile(string filepath)
+        {
+            if (string.IsNullOrWhiteSpace(filepath)) { throw new ArgumentNullException("filepath"); }
+            if (!File.Exists(filepath))
+            {
+                throw new FileNotFoundException("File not found", filepath);
+            }
+
+            FileInfo fi = new FileInfo(filepath);
+
+            var isExcludedQuery = from extension in GetExcludedExtensions()
+                                  where string.Compare(fi.Extension, extension, StringComparison.OrdinalIgnoreCase) == 0
+                                  select extension;
+            var isExcluded = isExcludedQuery.Count() > 0 ? true : false;
+
+            return !isExcluded;
+        }
+
+        /// <summary>
+        /// Creates a temporary file and returns the full path
+        /// </summary>
+        /// <param name="ensureFileDoesntExist">Wheter it is ensured that a file with the same name exists</param>
+        /// <param name="extension">Optional extension for the file</param>
+        /// <returns>Full path to the file created</returns>
+        public static string GetTempFilename(bool ensureFileDoesntExist, string extension = null)
```

Comment:
This is not used either.  It can also be completely replaced with `Path.ChangeExtension(Path.GetTempFileName() + ".", extension)` (I add the `+ "."` to retain the complete temp file name)

And as I said below, `Path.GetTempFileName` can be trusted to be unique.